Generator/Reader/XsdReader.php

```diff
@@ -0,0 +1,161 @@
+                $element->setName($node->getAttribute('name'))
+                        ->setElementAsFirst(true);
+                
+                $this->addChildrenElements($node, $element);
+
+                return $element;
+            }
+        }
+        return null;
+    }
+    
+    /**
+     * Add Children Elements to the given parentElement
+     * 
+     * @param DOMNode $parentNode
+     * @param Element $parentElement
+     */
+    protected function addChildrenElements($parentNode, &$parentElement) 
+    {
+        $children = $parentNode->childNodes;
+
+        foreach ($children as $node) {
+            if ($node->nodeType === XML_ELEMENT_NODE) {
+                switch ($node->localName) {
+                    case 'complexType':
+                    case 'all':
+                    case 'sequence':
+                        /**
+                         * Recursively looking at element(localName=element)node
+                         */
+                        $this->addChildrenElements($node, $parentElement);
+                        break;
+                    case 'element':
+                        if ($this->isComplexTypeNode($node)) {
+                            $element = new Reader\Xsd\ComplexTypeElement();
+                            $element->setName($node->getAttribute('name'));
+                            $this->setComplexTypeChildrenRecursively($node, $element);
+                            
+                            $parentElement->addChild($element);
+                        } else {
+                            $element = new Reader\Xsd\Element();
+                            $element->setName($node->getAttribute('name'));
+
+                            if ($node->getAttribute('type')) {
+                                $element->setDataType(DataMapper\DataTypeMapper::XsdToDto($node->getAttribute('type')));
+                            } else {
+                                $this->setElementsDataType($node, $element);
+                            }
+                            $parentElement->addChild($element);
+                        }
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+    }
+    
+    /**
+     * Recursively looks at an element which has DataType and sets the type of the element
+     * 
+     * @param DOMNode $node
+     * @param Element $element
+     */
+    protected function setElementsDataType($node, &$element)
+    {
+        $childNodes = $node->childNodes;
+        
+        foreach ($childNodes as $childNode) {
+            $localName = $childNode->localName;
+            if ($localName == 'simpleType') {
+                $this->setElementsDataType($childNode, $element);
+            } elseif ($localName == 'restriction' ) {
+                $element->setDataType(DataMapper\DataTypeMapper::XsdToDto($childNode->getAttribute('base')));
+                return;
+            } else {
```

Comment:
There is no need for this else statement. If the ```$localName``` is not matched in the previous clauses it will continue with the next iteration of the foreach anyway.